File: src/main/rest/repository/PurposeRepository.js
import Purpose from "../model/Purpose";

const PurposeRepository = {

   
    insert : (txn, purpose) => {
        return new Promise((resolve, reject) => {
    
            txn.executeSql(
                'insert into purpose(id, name, description, photo_url, disclosure_scope, start_date, end_date, stat) values(?,?,?,?,?,?,?,?)',
                [purpose.id, purpose.name, purpose.description, purpose.photoUrl, purpose.disclosureScope, purpose.startDate.toString(), purpose.endDate.toString(), purpose.stat],
                resolve,
                reject);
        })
    },

    selectById : (txn, id) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'select id, name, description, photo_url, disclosure_scope, start_date, end_date, stat from purpose where id = ?',
                [id],
                (res) => {
                   const row = res.rows.item(0);
                   resolve(new Purpose( row.id, row.name, row.description, row.photo_url, row.disclosure_scope, row.start_date, row.end_date, row.stat));
                },
                reject
            )
        })
    },

    selectAll : (txn) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                "select id, name, description, photo_url, disclosure_scope, start_date, end_date, stat from purpose",
                [],
                (res) => {
                    let data = [];
                    for(let i = 0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Purpose( row.id, row.name, row.description, row.photo_url, row.disclosure_scope, row.start_date, row.end_date, row.stat))
                    }
                    resolve(data);
                },
                reject
            );
        })
    },
    
    selectByStatIsActive : (txn) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                "select id, name, description, photo_url, disclosure_scope, start_date, end_date, stat from purpose where stat = 0",
                [],
                (res) => {
                    let data = [];
                    for(let i = 0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Purpose( row.id, row.name, row.description, row.photo_url, row.disclosure_scope, row.start_date, row.end_date, row.stat))
                    }
                    resolve(data);
                },
                reject
            )
        })
    },

    updatePurposeDate : (txn, id, purpose) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'update purpose set name = ?, description = ?, photo_url = ?, disclosure_scope = ?, start_date = ?, end_date = ?, stat = ? where id = ?',
                [purpose.name, purpose.description, purpose.photoUrl, purpose.disclosureScope, purpose.startDate.toString(), purpose.endDate.toString(), purpose.stat, id],
                (res) => {
                    resolve(res);
                },
                reject
            )
        })
    },

    updateStatById : (txn, id, stat) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'update purpose set stat = ? where id = ?',
                [stat, id],
                resolve,
                reject
            )
        })
    },

    deleteById : (txn, id) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'delete from purpose where id = ?',
                [id],
                resolve,
                reject
            )
        })
    },

    deleteAll : (txn) => {
        return new Promise((resolve, reject) => {
            txn.executeSql('delete from purpose', [] , resolve, reject);
        })
    }
}

export default PurposeRepository;

File: android/app/src/main/java/com/downfall/caterplanner/rest/repository/PurposeRepository.java
package com.downfall.caterplanner.rest.repository;

import android.content.ContentValues;
import android.database.Cursor;

import com.downfall.caterplanner.rest.model.Purpose;
import com.downfall.caterplanner.rest.db.SQLiteManager;
import com.downfall.caterplanner.rest.model.State;
import com.downfall.caterplanner.util.DateUtil;

import java.text.ParseException;

public class PurposeRepository extends BaseRepository {

    public long insert(Purpose purpose){
        ContentValues contentValues = new ContentValues();
        contentValues.put("id", purpose.getId());
        contentValues.put("name", purpose.getName());
        contentValues.put("description", purpose.getDescription());
        contentValues.put("image_url", purpose.getImageUrl());
        contentValues.put("disclosure_scope",purpose.getDisclosureScope());
        contentValues.put("start_at", purpose.getStartAt().toString());
        contentValues.put("decimal_day", purpose.getDecimalDay().toString());
        contentValues.put("stat", purpose.getStat().getValue());
        return db.insert("purpose", null, contentValues);
    }

    public Purpose selectById(long id) throws ParseException {
        final String sql =
                "select id, name, " +
                        "description, image_url, disclosure_scope, start_at, decimal_day, stat " +
                        "from purpose where id = ?";
        Cursor c = db.rawQuery(sql, new String[]{String.valueOf(id)});
        Purpose purpose = null;
        if(c.moveToFirst()){
            purpose = Purpose.builder()
                    .id(c.getLong(1))
                    .name(c.getString(2))
                    .description(c.getString(3))
                    .imageUrl(c.getString(4))
                    .disclosureScope( c.getInt(5))
                    .startAt(DateUtil.parseToDateTime(c.getString(6)))
                    .decimalDay(DateUtil.parseToDate(c.getString(7)))
                    .stat(State.findByValue(c.getInt(8)))
                    .build();

        }
        return purpose;
    }

    public Purpose[] selectByStatIsActive() throws ParseException {
        final String sql =
                "select id, name, " +
                        "description, image_url, disclosure_scope, start_at, decimal_day, stat " +
                        "from purpose where stat = 1";
        Cursor c = db.rawQuery(sql, null);
        Purpose[] purposes = new Purpose[c.getCount()];
        if(c.moveToNext()){
            purposes[c.getPosition()] =
                    Purpose.builder()
                    .id(c.getLong(1))
                    .name(c.getString(2))
                    .description(c.getString(3))
                    .imageUrl(c.getString(4))
                    .disclosureScope( c.getInt(5))
                    .startAt(DateUtil.parseToDateTime(c.getString(6)))
                    .decimalDay(DateUtil.parseToDate(c.getString(7)))
                    .stat(State.findByValue(c.getInt(8)))
                    .build();
        }
        return purposes.length == 0 ? null : purposes;
    }

    public void updatePurposeDate(long id , Purpose purpose){
        final String sql =
                "update purpose " +
                        "set name = ?, description = ?, image_url = ?, disclosure_scope = ?, " +
                        "start_at = ?, decimal_day = ?, stat = ?" +
                        "where id = ?";
        db.execSQL(sql,
                new String[]{
                    purpose.getName(), purpose.getDescription(), purpose.getImageUrl(),
                        String.valueOf(purpose.getDisclosureScope()), purpose.getStartAt().toString(), purpose.getDecimalDay().toString(),
                        String.valueOf(purpose.getStat().getValue())
                });
    }

    public void updateStatById(long id, State stat){
        final String sql =
                "update purpose " +
                        "set stat = ? " +
                        "where id = ?";
        db.execSQL(sql,
                new String[]{
                        String.valueOf(id),
                        String.valueOf(stat.getValue())
                });
    }

    public void deleteById(long id){
        final String sql =
                "delete from purpose where id = ?";
        db.execSQL(sql, new String[]{String.valueOf(id)});
    }



}
